UrbanEngine.Infrastructure/Migrations/20181002014956_InitialCreate.cs

```diff
@@ -0,0 +1,308 @@
+                });
+
+            migrationBuilder.CreateTable(
+                name: "UserRole",
+                schema: "public",
+                columns: table => new
+                {
+                    UserId = table.Column<long>(nullable: false),
+                    RoleId = table.Column<long>(nullable: false)
+                },
+                constraints: table =>
+                {
+                    table.PrimaryKey("PK_UserRole", x => new { x.UserId, x.RoleId });
+                    table.ForeignKey(
+                        name: "FK_UserRole_Role_RoleId",
+                        column: x => x.RoleId,
+                        principalSchema: "public",
+                        principalTable: "Role",
+                        principalColumn: "Id",
+                        onDelete: ReferentialAction.Cascade);
+                    table.ForeignKey(
+                        name: "FK_UserRole_User_UserId",
+                        column: x => x.UserId,
+                        principalSchema: "public",
+                        principalTable: "User",
+                        principalColumn: "Id",
+                        onDelete: ReferentialAction.Cascade);
+                });
+
+            migrationBuilder.InsertData(
+                schema: "public",
+                table: "Privilege",
+                columns: new[] { "Id", "Name" },
+                values: new object[] { 1L, "login" });
+
+            migrationBuilder.InsertData(
+                schema: "public",
+                table: "Role",
+                columns: new[] { "Id", "Name" },
+                values: new object[,]
+                {
+                    { 1L, "Site Administrator" },
+                    { 2L, "Events Manager" },
+                    { 3L, "CoWorking Night Workshop Organizer" },
+                    { 4L, "Member" }
+                });
+
+            migrationBuilder.CreateIndex(
+                name: "IX_Event_VenueId",
+                schema: "public",
+                table: "Event",
+                column: "VenueId");
+
+            migrationBuilder.CreateIndex(
+                name: "IX_RolePrivilege_PrivilegeId",
+                schema: "public",
+                table: "RolePrivilege",
+                column: "PrivilegeId");
+
+            migrationBuilder.CreateIndex(
+                name: "IX_TaggedUser_TagId",
+                schema: "public",
+                table: "TaggedUser",
+                column: "TagId");
+
+            migrationBuilder.CreateIndex(
+                name: "IX_User_CompanyId",
+                schema: "public",
+                table: "User",
+                column: "CompanyId");
+
+            migrationBuilder.CreateIndex(
+                name: "IX_UserRole_RoleId",
+                schema: "public",
+                table: "UserRole",
+                column: "RoleId");
+        }
+
+        protected override void Down(MigrationBuilder migrationBuilder)
```

Comment:
So with EF migrations, when I start the project it creates the tables, seeds the database, and is ready to start using the database. Is it going to re seed the same data every time I start it up? In Visual Studio I see that this method isn't called anywhere. What keeps us from seeding the data over and over again and prevent duplicate data during development/testing?

Reply:
There is an EF Migrations history table that keeps track of everything. When Migrate is called its only going to update you to the latest version. Each time you make changes to entities, seed data, etc, really anything that impacts your DbContext you add new migrations to be able to update to the next version